### PR TITLE
Chore: Moves common and response into separate packages

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
This PR is needed to avoid circular refrences when adding micro services [as described here](https://github.com/grafana/grafana/pull/30278#discussion_r557412182).

**Which issue(s) this PR fixes**:
Relates #29610

**Special notes for your reviewer**:

### CODE REVIEW

**Review by aknuds1 — COMMENTED** [review]
I'm reading your originating PR first, to understand the need.

File: pkg/api/utils/common.go [hunk 1]
```diff
@@ -30,30 +31,26 @@ func Wrap(action interface{}) macaron.Handler {
...
 // JSON creates a JSON response.
-func JSON(status int, body interface{}) *NormalResponse {
+func JSON(status int, body interface{}) *response.NormalResponse {
 	return Respond(status, body).Header("Content-Type", "application/json")
 }
 
-// jsonStreaming creates a streaming JSON response.
-func jsonStreaming(status int, body interface{}) streamingResponse {
+// JsonStreaming creates a streaming JSON response.
+func JsonStreaming(status int, body interface{}) response.StreamingResponse {
```
Comment (aknuds1) [inline pkg/api/utils/common.go:39]:
```suggestion
// JSONStreaming creates a streaming JSON response.
func JSONStreaming(status int, body interface{}) response.StreamingResponse {
```

**Review by aknuds1 — CHANGES_REQUESTED** [review]
Why not move the functions creating responses to the api/response package instead? Would be more logically coherent/self-documenting to me, and less ambiguous ("utils" is an ambiguous package name).

**Comment by hugohaggmark** [comment]
> Why not move the functions creating responses to the api/response package instead? Would be more logically coherent/self-documenting to me, and less ambiguous ("utils" is an ambiguous package name).

Thanks for the feedback, exactly which functions are you referring to?

**Comment by hugohaggmark** [comment]
So I moved Wrap into the routing package and the rest of the function from common into the response package. I hope that makes it clearer.

**Comment by aknuds1** [comment]
> > Why not move the functions creating responses to the api/response package instead? Would be more logically coherent/self-documenting to me, and less ambiguous ("utils" is an ambiguous package name).
> 
> Thanks for the feedback, exactly which functions are you referring to?
> I mean should everything in common.go be move back to response.go? We split common.go into 2 files recently and I was just respecting that split up.

I mean the functions creating responses, i.e. not Wrap.

I'll have a look at your latest revision.

**Review by marefr — APPROVED** [review]
Great :star2: LGTM

**Review by aknuds1 — APPROVED** [review]
LGTM

**Pull request merged**